app/controllers/health/health_controller.rb

```diff
@@ -3,10 +3,17 @@ class HealthController < AbstractHealthController
     private
 
     def health_checker
-      MultiHealthChecker.new(
+      checkers = {
         database: DatabaseHealthChecker,
-        workers: WorkerHealthChecker
-      )
+        workers: WorkerHealthChecker,
+      }
+      # Don't run worker health checks if we're not using workers (i.e. if the
+      # queue adapter is inline or async)
+      case Rails.application.config.active_job.queue_adapter
+      when :async, :inline
+        checkers.delete(:workers)
```

Comment:
Please add a test for this.

Reply:
Added 1bd72fcccb65959253a1e2c91e1d6a6afb1dcff8 and confirmed that it fails without the new health check code.